core/conn/jdbcT4/src/main/java/org/trafodion/jdbc/t4/TrafT4MultiQueriesPreparedStatement.java

```diff
@@ -0,0 +1,372 @@
+            int resultSetConcurrency) throws SQLException {
+        this(connection, sql, resultSetType, resultSetConcurrency, connection.holdability_, null);
+        if (connection.props_.t4Logger_.isLoggable(Level.FINE) == true) {
+            Object p[] = T4LoggingUtilities.makeParams(connection.props_, connection, sql, resultSetType,
+                    resultSetConcurrency);
+            connection.props_.t4Logger_.logp(Level.FINE, "TrafT4MultiQueriesPreparedStatement", "<init>", "", p);
+        }
+    }
+
+    TrafT4MultiQueriesPreparedStatement(TrafT4Connection connection, String sql, int resultSetType,
+            int resultSetConcurrency, int resultSetHoldability) throws SQLException {
+        this(connection, sql, resultSetType, resultSetConcurrency, resultSetHoldability, null);
+        if (connection.props_.t4Logger_.isLoggable(Level.FINE) == true) {
+            Object p[] = T4LoggingUtilities.makeParams(connection.props_, connection, sql, resultSetType,
+                    resultSetConcurrency, resultSetHoldability);
+            connection.props_.t4Logger_.logp(Level.FINE, "TrafT4MultiQueriesPreparedStatement", "<init>", "", p);
+        }
+
+    }
+
+    TrafT4MultiQueriesPreparedStatement(TrafT4Connection connection, String sql, int resultSetType,
+            int resultSetConcurrency, int resultSetHoldability, String stmtLabel) throws SQLException {
+        super(connection, sql, resultSetType, resultSetConcurrency, resultSetHoldability, stmtLabel);
+        if (connection.props_.t4Logger_.isLoggable(Level.FINE) == true) {
+            Object p[] = T4LoggingUtilities.makeParams(connection.props_, connection, sql, resultSetType,
+                    resultSetConcurrency, resultSetHoldability, stmtLabel);
+            connection.props_.t4Logger_.logp(Level.FINE, "TrafT4MultiQueriesPreparedStatement", "<init>", "", p);
+        }
+
+        if (resultSetType != ResultSet.TYPE_FORWARD_ONLY && resultSetType != ResultSet.TYPE_SCROLL_INSENSITIVE
+                && resultSetType != ResultSet.TYPE_SCROLL_SENSITIVE) {
+            throw TrafT4Messages.createSQLException(connection_.props_, connection_.getLocale(),
+                    "invalid_resultset_type", null);
+        }
+        if (resultSetConcurrency != ResultSet.CONCUR_READ_ONLY && resultSetConcurrency != ResultSet.CONCUR_UPDATABLE) {
+            throw TrafT4Messages.createSQLException(connection_.props_, connection_.getLocale(),
+                    "invalid_resultset_concurrency", null);
+        }
+        if ((resultSetHoldability != 0) && (resultSetHoldability != ResultSet.CLOSE_CURSORS_AT_COMMIT)
+                && (resultSetHoldability != ResultSet.HOLD_CURSORS_OVER_COMMIT)) {
+            throw TrafT4Messages.createSQLException(connection_.props_, connection_.getLocale(), "invalid_holdability",
+                    null);
+        }
+
+        sqlArr = sql.split(";");
```

Comment:
What happens if there is ';' in the character literal of the sql query?

Reply:
there will have syntax error.
eg: insert into tbl values ('a;a'); insert into tbl2 values ('b')
there will have 3 preparedStatments, each one do itself prepare, then there will throw exception when do insert into tbl values ('a
Actually this is a problem, i don't have a good solution now, maybe you can give some suggestions, thanks.

Reply:
You could search for the standalone ';' and ignore ';' when it is within a single quote.

Reply:
This seems very prone to opening application to sql code insertion attacks. We have to be VERY careful not to allow any literal string to be executed as sql code.

Reply:
@svarnau @selvaganesang actually, what TrafT4MultiQueriesPreparedStatement do is just TrafT4PreparedStatement do, so if TrafT4PreparedStatement can avoid insertion attacks or variables like $PARAM1, then TrafT4MultiQueriesPreparedStatement can

Reply:
I agree with you @mashengchen. But my suggestion is to take care of ';' when it appears in the midst of string literal in the param.  Do you think if that suggestion would hold good? If so, do you think it is better to make that change